lib/diagnostic.rb

```diff
@@ -108,27 +131,32 @@ def eat_meat(food)
 # which returns the string "kick, move, punch"
 
 class ComboAttack
-  attr_reader :moves, :damage
+  attr_reader :moves
+  attr_accessor :damage
 
   def initialize
     @moves = []
     @damage = 0
   end
 
+  def self.get_possible_moves
+    "kick, move, punch"
+  end
+
   def punch
-    @moves << 'punch'
-    @damage += 5
-    @damage *= multiplier
+    self.moves << 'punch'
+    self.damage += 5
+    self.damage *= multiplier
   end
 
   def move(direction)
-    @moves << "move #{direction}"
+    self.moves << "move #{direction}"
   end
 
   def kick
-    @moves << 'kick'
-    @damage += 10
-    @damage *= multiplier
+    self.moves << 'kick'
+    self.damage += 10
+    self.damage *= multiplier
```

Comment:
Not quite. The way to make the method chaining work is for `punch`, `kick`, and `move` to return `self`.That way, `ComboAttack.new.punch.move('left').kick.damage` is equivalent to

```ruby
combo = ComboAttack.new
combo.punch
combo.move('left')
combo.kick
combo.damage
```